etc/spack/defaults/config.yaml

```diff
@@ -74,6 +77,15 @@ config:
   dirty: false
 
 
+  # When set to true, concurrent instances of Spack will use locks to
+  # avoid modifying the install tree, database file, etc. If false, Spack
+  # will disable all locking, but you must NOT run concurrent instances
+  # of Spack.  For filesystems that don't support locking, you should set
+  # this to false and run one Spack at a time, but otherwise we recommend
+  # enabling locks.
+  locks: true
```

Comment:
I mean, this is what the locks are for 😄 If there was a good way to do this, we wouldn't need locks.

Reply:
Well, right now we are (cleverly, and thanks to your lock implementation) using `flock` features to permit concurrent use of Spack from different users. I was wondering if we can have a simpler global lock that doesn't expose users to potential races, and doesn't use `flock`. The idea is to exit with a message like:
```console
Spack is currently being used by somebody else. Retry later.
```
if the lock is already taken, instead of saying to users that they are on their own.

Reply:
Although it's probably not applicable here: last time we encountered a FS without lock capabilitities (Piz Daint), I just changed the logic to created locks for pseudo-files on another shared filesystem of the machine...

For Spack `locks: false` in the `SPACK_ROOT`, one could also think about setting a path for locks of shared pseudo-files... not sure it it's worth the effort (I did not do this with spack but with some post-processing).